tracker-pt/point_extractor.cpp

```diff
@@ -19,6 +19,108 @@
 #include <algorithm>
 #include <cinttypes>
 
+
+
+/*
+http://en.wikipedia.org/wiki/Mean-shift
+In this application the idea, is to eliminate any bias of the point estimate 
+which is introduced by the rather arbitrary thresholded area. One must recognize
+that the thresholded area can only move in one pixel increments since it is
+binary. Thus, its center of mass might make "jumps" as pixels are added/removed
+from the thresholded area.
+With mean-shift, a moving "window" or kernel is multiplied with the gray-scale
+image, and the COM is calculated of the result. This is iterated where the
+kernel center is set the previously computed COM. Thus, peaks in the image intensity
+distribution "pull" the kernel towards themselves. Eventually it stops moving, i.e.
+then the computed COM coincides with the kernel center. We hope that the 
+corresponding location is a good candidate for the extracted point.
+The idea similar to the window scaling suggested in  Berglund et al. "Fast, bias-free 
+algorithm for tracking single particles with variable size and shape." (2008).
+*/
+static cv::Vec2d MeanShiftIteration(const cv::Mat &frame_gray, cv::Vec2d &current_center, double filter_width)
+{
+    const double s = 1.0/filter_width;
+    
+    auto EvalFilter = [=](int row, int col) -> double
+    {
+        double dx = (col - current_center[0])*s;
+        double dy = (row - current_center[1])*s;
+        double f = std::max(0.0, 1.0 - dx*dx - dy*dy);
+        return f*f;
+    };
+
+    double m = 0;
+    cv::Vec2d com(0.0, 0.0);
+    for (int i=0; i < frame_gray.rows; i++)
+    {
+        auto frame_ptr = (uint8_t *)frame_gray.ptr(i);
+        for (int j=0; j < frame_gray.cols; j++)
+        {
+            double val = frame_ptr[j] / 255.;
+            val *= EvalFilter(i, j);
+            val = val*val; // taking the square wights brighter parts of the image stronger.
+            m      += val;
+            com[0] += j * val;
+            com[1] += i * val; 
+        }
+    }
+    com *= 1.0/m;
+    return com;
+}
+
+/*  The mean-shift iteration is sensitive to nearby blobs, not belonging to the central blob the
+    position of which we want to find. This function attempts to remove the intensity peak pertaining
+    to such nearby blobs.
+
+    It computes a discrete voronoi diagram based on the distances from thresholded blobs (binary_inv).
+    Then it fills the image area of voronoi regions pertaining to other blobs with the average 
+    brightness found surrounding the central blob.
+
```

Comment:
The sheer complexity, and 1.2 megs per each float matrix like these below.

How much CPU usage at 187 Hz in 320x240? How much CPU usage at 75 Hz in 640x480? What CPU?